src/x64dbg-module/SnowmanView.cpp

```diff
@@ -0,0 +1,198 @@
+#include <nc/config.h>
+#include <nc/common/Branding.h>
+#include <nc/common/Foreach.h>
+#include <nc/common/make_unique.h>
+#include <nc/common/Types.h>
+#include <nc/core/image/ByteSource.h>
+#include <nc/core/Context.h>
+#include <nc/core/arch/Instruction.h>
+#include <nc/core/image/Image.h>
+#include <nc/core/mangling/Demangler.h>
+#include <nc/core/image/Section.h>
+#include <nc/gui/MainWindow.h>
+#include <nc/gui/Project.h>
+#include <nc/gui/InstructionsView.h>
+#include <nc/gui/CxxView.h>
+
+#include "SnowmanView.h"
+#include <QVBoxLayout>
+#include <QMessageBox>
+#include <QTreeView>
+#include <QMenuBar>
+#include <QPlainTextEdit>
+
+extern "C" __declspec(dllexport) SnowmanView* CreateSnowman(QWidget* parent)
+{
+    return new SnowmanView(parent);
+}
+
+extern "C" __declspec(dllexport) void DecompileAt(SnowmanView* snowman, duint start, duint end)
+{
+    snowman->decompileAt(start, end);
+}
+
+extern "C" __declspec(dllexport) void CloseSnowman(SnowmanView* snowman)
+{
+    snowman->close();
+}
+
+class DbgByteSource : public nc::core::image::ByteSource
+{
+    virtual nc::ByteSize readBytes(nc::ByteAddr addr, void *buf, nc::ByteSize size) const
+    {
+        if(DbgMemRead(addr, (unsigned char*)buf, size))
+            return size;
+        else
+            return 0;
+    }
+};
+
+class DbgDemangler: public nc::core::mangling::Demangler {

```

Comment:
BTW, you probably do not need this class: we have a default demangler that does something sensible for common cases (GCC, MSVC).